File: Assignment_33/Program2.cpp
/*
Write a program which accepts two number from user and display position of common ON bits from that two numbers.
Input : 10    15
Output : 2     4
*/

#include<iostream>
using namespace std;

typedef unsigned int UINT;

void CommonBits(UINT iNo1,UINT iNo2)
{
    UINT iPos = 1;
    UINT iCommon = 0;

    iCommon = iNo1 & iNo2;

    cout<<"Common ON bit positions are:";

    while(iCommon != 0)
    {
        if(iCommon & 1)
        {
            cout<<iPos<<" ";
        }
        iPos++;
        iCommon = iCommon >> 1;
    }
}

int main()
{
    UINT iValue1 = 0,iValue2 = 0;

    cout<<"Enter the First Number:";
    cin>>iValue1;

    cout<<"Enter the Second Number:";
    cin>>iValue2;

    CommonBits(iValue1,iValue2);    

    return 0;
}

File: Assignment_50/Program2.java
/* Write a java program which accepts string from user and display below pattern.
Input :  iRow = 4   iCol = 4
Output : *  *  *  #
         *  *  #  @
         *  #  @  @
         #  @  @  @
*/

import java.util.*;

class Pattern
{
    public void Pattern(int iRow,int iCol)
    {
        int i = 0,j = 0;

        for(i = 1;i <= iRow;i++)
        {
            for(j = 1;j <= iCol;j++)
            {
                if(i + j == iRow + 1)
                {
                    System.out.print("#\t");
                }
                else if(i + j < iRow + 1)
                {
                    System.out.print("*\t");
                }
                else
                {
                    System.out.print("@\t");
                }
            }
            System.out.println();
        }
    }
}

class Program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the no. of Rows : ");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter the no. of Columns : ");
        int iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Pattern(iValue1,iValue2);
    }
}

